### DISCUSSION TITLE
Set Imagedescription in TIFF save...

### DISCUSSION THREAD

**Discussion by carlkesselman** [libvips/libvips#2050] [Q&A]
Ask a question...
Hi,

I know it is possible to write vips metadata to the image description tag in a TIFF output. However, I would like to provide my own value for this field, specifically the XML that describes an OME-TIFF file.  Is it possible to somehow override the behavior of the properties value to provide this?

Thanks,

Carl

**Comment by jcupitt** [comment]
Hi @carlkesselman,

Yes, set the `image-description` metadata item:

https://libvips.github.io/libvips/API/current/libvips-header.html#VIPS-META-IMAGEDESCRIPTION:CAPS

and don't set the `properties` option.

**Comment by jcupitt** [comment]
You must `vips_copy()` the image before modifying the metadata to make sure you have a private copy. In Python, for example:

```python
image = image.copy()
image.set('image-description', 'banana')
image.write_to_file('x.tif')
```

Will write a file with `TIFFTAG_IMAGEDESCRIPTION` set to `banana`.

**Comment by carlkesselman** [comment]
Perfect.  Thanks!

Carl


----------------------------------------------------------

Dr. Carl Kesselman
Dean’s Professor, Epstein Department of Industrial and Systems Engineering
Fellow, Information Sciences Institute Viterbi School of Engineering Professor,

Preventive Medicine Keck School of Medicine

University of Southern California
4676 Admiralty Way, Suite 1001, Marina del Rey, CA  90292-6695
Phone:  +1 (310) 448-9338
 Email:  carl@isi.edu
Web: http://www.isi.edu/~carl
On Aug 31, 2020, 9:50 AM -0700, John Cupitt <notifications@github.com>, wrote:

You must vips_copy() the image before modifying the metadata to make sure you have a private copy. In Python, for example:

image = image.copy()
image.set('image-description', 'banana')
image.write_to_file('x.tif')

Will write a file with TIFFTAG_IMAGEDESCRIPTION set to 'banana'.
